thrust/thrust/optional.h

```diff
@@ -2776,10 +2776,11 @@ class optional<T&>
   template <class... Args>
   _CCCL_HOST_DEVICE T& emplace(Args&&... args) noexcept
   {
-    static_assert(std::is_constructible<T, Args&&...>::value, "T must be constructible with Args");
+    static_assert(sizeof...(Args) == 1, "Args must be a single lvalue");
+    static_assert(std::is_lvalue_reference<Args...>::value, "Args must be a single lvalue");
```

Comment:
I would rather change the function signature to just take a `_Arg&` instead of variadics

Reply:
That was my initial preference, but I didn't want to change the interface. I guess functionally the change would still be the same as changing the interface anyway, so it wouldn't matter technically.